Live/Classes/MainScene.cpp

```diff
@@ -98,11 +108,35 @@ void MainScene::update(float delta) {
         isKeyPressed(EventKeyboard::KeyCode::KEY_S)) {
         m_player->moveY(-(MOVE_STEP * delta));
     }
+    if (isKeyPressed(EventKeyboard::KeyCode::KEY_Z)) {
+        for (auto it : m_items) {
+            if (m_player->pickup(it)) {
+                m_items.erase(std::remove(m_items.begin(), m_items.end(), it));
+                std::cerr << m_items.size() << std::endl;
+                break;  // Only allow one pick up at a time
+            }
+        }
+    }
 
-    // Update the HUD
+    for (int i = 1; i <= 10; i++) {
+        int code = static_cast<int>(EventKeyboard::KeyCode::KEY_0) + i;
+        if (isKeyPressed(static_cast<EventKeyboard::KeyCode>(code))) {
+            if (isKeyPressed(EventKeyboard::KeyCode::KEY_X)) {
+                Item* item = m_player->drop(i);
+                if (item != NULL) {
+                    Vec2 pos = m_player->getPosition();
+                    m_items.push_back(item);
+                    addChild(item->newSprite());
+                    item->setPosition(pos.x, pos.y);
```

Comment:
doesn't doing `item->setPosition(m_player->getPosition())` work? I thought you could also set the position by directly passing in a Vec2, check though.

Reply:
Done, uodated gameObject